src/python/pants/backend/jvm/tasks/scalafmt.py

```diff
@@ -51,15 +58,46 @@ def invoke_tool(self, _, target_sources):
     # If no config file is specified use default scalafmt config.
     config_file = self.get_options().configuration
     args = list(self.additional_args)
-    args.extend(['--files', ','.join(source for _, source in target_sources)])
-    if config_file != None:
+    if config_file is not None:
       args.extend(['--config', config_file])
+    if self.get_options().output_dir:
+      args.append('--stdout')
+
+    result = 0
+    created_dirs = set()
+
+    for _, source in target_sources:
+      res, workunit = self.runjava(classpath=self.tool_classpath('scalafmt'),
+                                   main='org.scalafmt.cli.Cli',
+                                   args=args + ['--files', source],
+                                   workunit_name='scalafmt',
+                                   jvm_options=self.get_options().jvm_options,
+                                   return_workunit=True)
+      result |= res
+
+      if self.get_options().output_dir is not None:
+        with open(workunit.output_paths()['stdout'], 'r') as f:
+          formatted_file = f.read()
+
+        with open(source, 'r') as f:
+          unformatted_file = f.read()
+
+        if formatted_file != unformatted_file:
```

Comment:
It seems like odd semantics to only write the changed files. Why is this the requirement?